File: backend/app/schemas/digest.py
"""
StandupBot — Digest Schemas

Request/response schemas for digest views and analytics.
"""

from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, Field


class DigestMemberEntry(BaseModel):
    """A member's entry within a digest."""

    member_id: UUID
    member_name: str
    submitted: bool
    has_blocker: bool
    answers: list[dict] = []


class DigestResponse(BaseModel):
    """Full digest response."""

    id: UUID
    team_id: UUID
    digest_date: date
    ai_summary: str | None = None
    total_members: int
    responded_count: int
    response_rate: float = Field(..., description="Response rate as percentage (0-100)")
    non_responders: list[str] = Field(default_factory=list, description="Names of non-responders")
    blockers: list[dict] = Field(default_factory=list, description="Flagged blockers")
    entries: list[DigestMemberEntry] = []
    status: str
    sent_at: datetime | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class DigestListItem(BaseModel):
    """Summary digest for list/history views."""

    id: UUID
    digest_date: date
    total_members: int
    responded_count: int
    response_rate: float
    status: str
    sent_at: datetime | None = None

    model_config = {"from_attributes": True}


class TeamStatsResponse(BaseModel):
    """Dashboard analytics response for a team."""

    team_id: UUID
    period_days: int = 30
    average_response_rate: float
    total_standups: int
    total_blockers: int
    member_stats: list[dict] = Field(default_factory=list, description="Per-member response rates")


class BlockerTrendResponse(BaseModel):
    """Blocker frequency trends."""

    team_id: UUID
    period_days: int = 30
    total_blockers: int
    trend_data: list[dict] = Field(default_factory=list, description="Daily blocker counts")
    common_themes: list[str] = Field(default_factory=list, description="Most frequent blocker themes")

File: frontend/src/types/digest.ts
/**
 * StandupBot — Digest Types
 */

export interface DigestMemberEntry {
  member_id: string;
  member_name: string;
  submitted: boolean;
  has_blocker: boolean;
  answers: { question: string; answer: string }[];
}

export interface Digest {
  id: string;
  team_id: string;
  digest_date: string;
  ai_summary: string | null;
  total_members: number;
  responded_count: number;
  response_rate: number;
  non_responders: string[];
  blockers: { member: string; text: string }[];
  entries: DigestMemberEntry[];
  status: string;
  sent_at: string | null;
  created_at: string;
}

export interface DigestListItem {
  id: string;
  digest_date: string;
  total_members: number;
  responded_count: number;
  response_rate: number;
  status: string;
  sent_at: string | null;
}

export interface TeamStats {
  team_id: string;
  period_days: number;
  average_response_rate: number;
  total_standups: number;
  total_blockers: number;
  member_stats: { member_id: string; name: string; response_rate: number }[];
}

export interface BlockerTrend {
  team_id: string;
  period_days: number;
  total_blockers: number;
  trend_data: { date: string; count: number }[];
  common_themes: string[];
}
